mlrun/datastore/base.py

```diff
@@ -699,3 +700,26 @@ def _validate_https_token(self):
                 f"A AUTH TOKEN should not be provided while using {self._schema} "
                 f"schema as it is not secure and is not recommended."
             )
+
+
+class DsSanitizer:
+    def __init__(self, cls, *args, **kwargs):
+        if not issubclass(cls, fsspec.AbstractFileSystem):
+            raise ValueError("Class must be a subclass of fsspec.AbstractFileSystem")
+
+        class DsSanitizerTemp(cls):
+            @classmethod
+            def _strip_protocol(cls, url):
+                if url.startswith("ds://"):
+                    parsed_url = urlparse(url)
+                    url = parsed_url.path[1:]
```

Comment:
Are we expecting a situation where we have something like `ds://` here at all? IIUC, this is created by the stores themselves, which should already get the url without a `ds://` prefix. Maybe I'm missing something.
Also, why calling `super()._strip_protocol()` as well? Are paths like `ds://s3://...` possible?

Reply:
Yes: 
ingest()->featureset.purge_targets()-> driver.purge()->self._get_store().rm(self.get_target_path(), recursive=True)